ci_tests/src/index.ts

```diff
@@ -1,49 +1,122 @@
 import {
   CONTRACTS,
   ChainId,
+  ChainName,
   getEmitterAddressEth,
   getSignedVAAWithRetry,
   parseSequenceFromLogEth,
+  postVaaSolana,
 } from "@certusone/wormhole-sdk";
 import { WormholeRelayer__factory } from "@certusone/wormhole-sdk/lib/cjs/ethers-contracts";
 import {
   getDeliveryHashFromLog,
   getWormholeLog,
 } from "@certusone/wormhole-sdk/lib/cjs/relayer";
-import { Networkish } from "@ethersproject/networks";
+import { NodeWallet } from "@certusone/wormhole-sdk/lib/cjs/solana";
+import { PostedMessageData } from "@certusone/wormhole-sdk/lib/cjs/solana/wormhole";
+import { BN, web3 } from "@coral-xyz/anchor";
 import { NodeHttpTransport } from "@improbable-eng/grpc-web-node-http-transport";
-import { Wallet, getDefaultProvider, utils } from "ethers";
-import { existsSync, writeFileSync } from "fs";
+import * as spl from "@solana/spl-token";
+import {
+  BigNumber,
+  ContractReceipt,
+  ContractTransaction,
+  Wallet,
+  providers,
+  utils,
+} from "ethers";
 import { DummyTokenMintAndBurn__factory } from "../evm_binding/factories/DummyToken.sol/DummyTokenMintAndBurn__factory";
 import { DummyToken__factory } from "../evm_binding/factories/DummyToken.sol/DummyToken__factory";
 import { ERC1967Proxy__factory } from "../evm_binding/factories/ERC1967Proxy__factory";
 import { NttManager__factory } from "../evm_binding/factories/NttManager__factory";
 import { TransceiverStructs__factory } from "../evm_binding/factories/TransceiverStructs__factory";
 import { TrimmedAmountLib__factory } from "../evm_binding/factories/TrimmedAmount.sol/TrimmedAmountLib__factory";
 import { WormholeTransceiver__factory } from "../evm_binding/factories/WormholeTransceiver__factory";
+import { NTT } from "../solana_binding/ts/sdk";
 
 // Chain details to keep track of during the testing
-interface ChainDetails {
+type ChainDetails = EVMChainDetails | SolanaChainDetails;
+interface EVMChainDetails extends BaseDetails {
+  type: "evm";
+  signer: Wallet;
+}
+interface SolanaChainDetails extends BaseDetails {
+  type: "solana";
+  signer: web3.Keypair;
+}
+interface BaseDetails {
   chainId: number;
+  chainName: ChainName;
   transceiverAddress: string;
   managerAddress: string;
   NTTTokenAddress: string;
   wormholeCoreAddress: string;
-  rpcEndpoint: Networkish;
 }
 
 const ETH_PRIVATE_KEY =
   "0x4f3edf983ac636a65a842ce7c78d9aa706d3b113bce9c46f30d7d21715b23b1d"; // Ganache default private key
 const ETH_PUBLIC_KEY = "0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1";
+const ETH_SIGNER = new Wallet(
+  ETH_PRIVATE_KEY,
+  new providers.JsonRpcProvider("http://eth-devnet:8545")
+);
+const BSC_SIGNER = new Wallet(
+  ETH_PRIVATE_KEY,
+  new providers.JsonRpcProvider("http://eth-devnet2:8545")
+);
+const SOL_PRIVATE_KEY = web3.Keypair.fromSecretKey(
+  new Uint8Array([
```

Comment:
Anyway we could pretty this format up? A bunch of numbers in an array is kinda ugly to look at.